### ISSUE TITLE
Batch evolve

### ISSUE THREAD

**Issue by SimonMoua** [vinnymac/PokeNurse#182]
Thanks to your update, now the transfer are correctly done in batch. But so is the evolve. While it may be technically possible to do, it's more risky, because the UI in the game only allow an evolve every 22 sec (these accounts may have more chance of getting banned in the future).

Hopefully, you could take the current evolve logic, do a for each pokemon selected, evolve, wait 22 sec, update the loading bar (no need for the count, as it was source of other problems before). 

If it's not possible without re-introducing the bug you took out by doing the batch, then ignore this, but currently, it does not simulate the app, and I'm now too scared to use this program for evolves.

**Comment by vinnymac** [comment]
@SimonMoua the issue wasn't with our app as much as it was with pogobuf + the PoGo API. I couldn't control the errors being thrown from the API whenever an evolve would fail for seemingly no reason. Due to these issues I decided to do away with the delays and evolves and use the simpler batch logic which works a lot more consistently (even if it does still throw errors). It is just much more likely to actually cause the transfer/evolves the user requested. The app is already doing things that the real app might not do, and also not doing things the real app might do, so if you are concerned about them being able to 'detect the difference'. I would say they have always been able to do that if they really wanted to. However I think they are slightly too busy to be worrying about all third party tools, and they have other kinds of protections. Thanks to their handy work we now have to implement Paid Hashing Keys and the like to keep users secure.

TL;DR If using the app makes you nervous I would really stay away from all third party tools because they all have a risk at the end of the day.
